readme.txt
go with a admin console to the path where the program is unpacked.

enter the following command: regsvr32 MSCOMM32.OCX

hit enter and start the program

to enable the support for ACD you need to add the following to the platform.txt of arduino:


################################################################################################
#             Stuff for ACD (Auto Connect Disconnect) by the SerialConsole program             #
################################################################################################
recipe.hooks.deploy.postupload.pattern=F:\Github\SerialConsole\ZebroMote.exe "OC {serial.port}"
recipe.hooks.deploy.preupload.pattern=F:\Github\SerialConsole\ZebroMote.exe "CC {serial.port}"
recipe.hooks.deploy.errorupload.pattern=F:\Github\SerialConsole\ZebroMote.exe "CC {serial.port}"
################################################################################################


platform.txt can be found in the following places FOR VISUAL MICRO:
    C:\Users\<username>\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.16
OR FOR ARDUINO:
	C:\Program Files (x86)\Arduino\hardware\arduino\avr





changelog.txt
Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui.

Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt. Duis leo. Sed fringilla mauris sit amet nibh. Donec sodales sagittis magna. Sed consequat, leo eget bibendum sodales, augue velit cursus nunc, quis gravida magna mi a libero. Fusce vulputate eleifend sapien. Vestibulum purus quam, scelerisque ut, mollis sed, nonummy id, metus. Nullam accumsan lorem in dui. Cras ultricies mi eu turpis hendrerit fringilla. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; In ac dui quis mi consectetuer lacinia.

Nam pretium turpis et arcu. Duis arcu tortor, suscipit eget, imperdiet nec, imperdiet iaculis, ipsum. Sed aliquam ultrices mauris. Integer ante arcu, accumsan a, consectetuer eget, posuere ut, mauris. Praesent adipiscing. Phasellus ullamcorper ipsum rutrum nunc. Nunc nonummy metus. Vestibulum volutpat pretium libero. Cras id dui. Aenean ut eros et nisl sagittis vestibulum. Nullam nulla eros, ultricies sit amet, nonummy id, imperdiet feugiat, pede. Sed lectus. Donec mollis hendrerit risus. Phasellus nec sem in justo pellentesque facilisis. Etiam imperdiet imperdiet orci. Nunc nec neque. Phasellus leo dolor, tempus non, auctor et, hendrerit quis, nisi.

Curabitur ligula sapien, tincidunt non, euismod vitae, posuere imperdiet, leo. Maecenas malesuada. Praesent congue erat at massa. Sed cursus turpis vitae tortor. Donec posuere vulputate arcu. Phasellus accumsan cursus velit. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Sed aliquam, nisi quis porttitor congue, elit erat euismod orci, ac placerat dolor lectus quis orci. Phasellus consectetuer vestibulum elit. Aenean tellus metus, bibendum sed, posuere ac, mattis non, nunc. Vestibulum fringilla pede sit amet augue. In turpis. Pellentesque posuere. Praesent turpis.

Aenean posuere, tortor sed cursus feugiat, nunc augue blandit nunc, eu sollicitudin urna dolor sagittis lacus. Donec elit libero, sodales nec, volutpat a, suscipit non, turpis. Nullam sagittis. Suspendisse pulvinar, augue ac venenatis condimentum, sem libero volutpat nibh, nec pellentesque velit pede quis nunc. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Fusce id purus. Ut varius tincidunt libero. Phasellus dolor. Maecenas vestibulum mollis diam. Pellentesque ut neque. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.

In dui magna, posuere eget, vestibulum et, tempor auctor, justo. In ac felis quis tortor malesuada pretium. Pellentesque auctor neque nec urna. Proin sapien ipsum, porta a, auctor quis, euismod ut, mi. Aenean viverra rhoncus pede. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Ut non enim eleifend felis pretium feugiat. Vivamus quis mi. Phasellus a est. Phasellus magna.

In hac habitasse platea dictumst. Curabitur at lacus ac velit ornare lobortis. Curabitur a felis in nunc fringilla tristique. Morbi mattis ullamcorper velit. Phasellus gravida semper nisi. Nullam vel sem. Pellentesque libero tortor, tincidunt et, tincidunt eget, semper nec, quam. Sed hendrerit. Morbi ac felis. Nunc egestas, augue at pellentesque laoreet, felis eros vehicula leo, at malesuada velit leo quis pede. Donec interdum, metus et hendrerit aliquet, dolor diam sagittis ligula, eget egestas libero turpis vel mi. Nunc nulla. Fusce risus nisl, viverra et, tempor et, pretium in, sapien. Donec venenatis vulputate lorem.

Morbi nec metus. Phasellus blandit leo ut odio. Maecenas ullamcorper, dui et placerat feugiat, eros pede varius nisi, condimentum viverra felis nunc et lorem. Sed magna purus, fermentum eu, tincidunt eu, varius ut, felis. In auctor lobortis lacus. Quisque libero metus, condimentum nec, tempor a, commodo mollis, magna. Vestibulum ullamcorper mauris at ligula. Fusce fermentum. Nullam cursus lacinia erat. Praesent blandit laoreet nibh.

Fusce convallis metus id felis luctus adipiscing. Pellentesque egestas, neque sit amet convallis pulvinar, justo nulla eleifend augue, ac auctor orci leo non est. Quisque id mi. Ut tincidunt tincidunt erat. Etiam feugiat lorem non metus. Vestibulum dapibus nunc ac augue. Curabitur vestibulum aliquam leo. Praesent egestas neque eu enim. In hac habitasse platea dictumst. Fusce a quam. Etiam ut purus mattis mauris sodales aliquam. Curabitur nisi. Quisque malesuada placerat nisl. Nam ipsum risus, rutrum vitae, vestibulum eu, molestie vel, lacus.


GlobalFunctions.bas
Attribute VB_Name = "GlobalFunctions"
Option Explicit


Public Declare Function GetTickCount Lib "kernel32" () As Long


Sub Main()
    Dim commandProcessed As String
    
    uDontDrawDots = True
    
    commandProcessed = Replace(Command, Chr(34), "")
    
    If Len(commandProcessed) > 0 Then
        MsgBox "'" & commandProcessed & "'"
        SendMessageToInstances commandProcessed
    Else
        frmMain.Show
    End If
    
    
End Sub




Public Sub wait(ByVal dblMilliseconds As Double)
    Dim dblStart As Double
    Dim dblEnd As Double
    Dim dblTickCount As Double
    
    dblTickCount = GetTickCount()
    dblStart = GetTickCount()
    dblEnd = GetTickCount + dblMilliseconds
    
    Do
    DoEvents
    dblTickCount = GetTickCount()
    Loop Until dblTickCount > dblEnd Or dblTickCount < dblStart
       
    
End Sub



MultipleInstances.bas
Attribute VB_Name = "MultipleInstances"
Private Declare Function FindWindow Lib "user32" Alias "FindWindowA" (ByVal lpClassName As String, ByVal lpWindowName As String) As Long
Private Declare Function PostMessage Lib "user32" Alias "PostMessageA" (ByVal hwnd As Long, ByVal wMsg As Long, ByVal wParam As Long, lParam As Any) As Long
Private Declare Function GetClassName Lib "user32" Alias "GetClassNameA" (ByVal hwnd As Long, ByVal lpClassName As String, ByVal nMaxCount As Long) As Long
Private Declare Function ShowWindow Lib "user32" (ByVal hwnd As Long, ByVal nCmdShow As Long) As Long

Private Declare Function FindWindowEx Lib "user32.dll" Alias "FindWindowExA" (ByVal hWnd1 As Long, ByVal hWnd2 As Long, ByVal lpsz1 As String, ByVal lpsz2 As String) As Long
Private Declare Function SendMessageByString Lib "user32" Alias "SendMessageA" (ByVal hwnd As Long, ByVal wMsg As Long, ByVal wParam As Long, ByVal lParam As String) As Long

Private Const SW_SHOWNORMAL = 1
Private Const WM_CLOSE = &H10
Private Const WM_SETTEXT = &HC


Sub SendMessageToInstances(whatToSend As String)
    Dim WinWnd As Long, Ret As String, RetVal As Long, lpClassName As String
    Dim textboxHandle As Long
    
    Dim strSplit() As String
    strSplit = Split(whatToSend, " ")
    
    If UBound(strSplit) <> 1 Then
        MsgBox "Possible commands:" & vbCrLf & vbCrLf & "CC COM4" & vbCrLf & "OC COM5"
        Exit Sub
    End If
    
    'Ask for a Window title
    Ret = strSplit(1) & " - SerialConsole - V1.0 by Ricardo de Roode" 'InputBox("Enter the exact window title:" + Chr$(13) + Chr$(10) + "Note: must be an exact match")
    'Search the window
    WinWnd = FindWindow(vbNullString, Ret)
    If WinWnd = 0 Then
        'MsgBox "no window"
        Exit Sub
    End If
    
    textboxHandle = FindWindowEx(WinWnd, 0&, "ThunderRT6TextBox", vbNullString)
    
    If textboxHandle = 0 Then
        'MsgBox "no textbox"
        Exit Sub
    End If
    
    SendMessageByString textboxHandle, WM_SETTEXT, 0&, whatToSend
    
    'Show the window
    'ShowWindow WinWnd, SW_SHOWNORMAL
    'Create a buffer
    'lpClassName = Space(256)
    'retrieve the class name
    'RetVal = GetClassName(WinWnd, lpClassName, 256)
    'Show the classname
    'MsgBox "Classname: " + Left$(lpClassName, RetVal)
    'Post a message to the window to close itself
    'PostMessage WinWnd, WM_CLOSE, 0&, 0&
End Sub


Arduino_test/Ansiterm.h
/* 
Ansiterm.cpp -- an Arduino library that simplifies using the ANSI terminal
control escape sequences. This allows your code to  position text exactly in the
terminal, to add colour and other effects. To use Ansiterm, an ANSI terminal must
be connected to the Arduino's serial port. The terminal built into the Arduino 
IDE does not respond to the ANSI codes, but most MacOs and Linux terminals do. 
In Windows, I read that Hyperterm responds to ANSI codes. Realterm responds to most
codes, but will not change the color of the foreground.

The accompanying example file illustrates the library's use.

Copyright (c) 2009 Bruce Robertson, VE9QRP. All rights reserved.

This library is free software; you can redistribute it and/or modify it under
the terms of the GNU Lesser General Public License as published by the Free
Software Foundation; either version 2.1 of the License, or (at your option)
any later version.

This library is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
details.

You should have received a copy of the GNU Lesser General Public License along
with this library; if not, write to the Free Software Foundation, Inc., 51
Franklin St, Fifth Floor, Boston, MA 02110-1301 USA

The latest version of this software is available at http://code.google.com/p/qrptracker/
*/

#ifndef Ansiterm_h
#define Ansiterm_h

#include "Arduino.h"

#define ESCAPE 0x1B
#define BRACE '['
#define BLACK 0
#define RED 1
#define      GREEN 2
#define YELLOW 3
#define BLUE 4
#define MAGENTA 5
#define CYAN 6 
#define WHITE 7
#define BOLD_ON 1
#define BOLD_OFF 22
#define UNDERLINE_ON 4
#define UNDERLINE_OFF 24
#define ITALICS_ON 3
#define ITALICS_OFF 23
#define STRIKETHROUGH_ON 9
#define STRIKETHROUGH_OFF 29
#define INVERSE_ON 7
#define INVERSE_OFF 27
#define RESET 1
#define DEFAULT_FOREGROUND 39
#define DEFAULT_BACKGROUND 49


class Ansiterm {
     
public:
     Ansiterm();
     void home();
     void xy(int x, int y);
     void up(int x); 
     void down(int x);
     void forward(int x);
     void backward(int x);
     void eraseLine();
     void eraseScreen();
     void setBackgroundColor(int color);
     void setForegroundColor(int color);
       void boldOn();
       void boldOff();
       void underlineOn();
       void underlineOff();
       void italicsOn();
       void italicsOff();
       void strikethroughOn();
       void strikethroughOff();
       void inverseOn();
       void inverseOff();
       void reset();
       void defaultBackground();
       void defaultForeground();
       void fill(int x1, int y1, int x2, int y2);
private:
     void preamble();
     void preambleAndNumberAndValue(int x, char v);
     void setAttribute(int a);
     
};
#endif

Arduino_test/Ansiterm.cpp

/* 
Ansiterm.cpp -- an Arduino library that simplifies using the ANSI terminal
control escape sequences. This allows your code to  position text exactly in the
terminal, to add colour and other effects. To use Ansiterm, an ANSI terminal must
be connected to the Arduino's serial port. The terminal built into the Arduino 
IDE does not respond to the ANSI codes, but most MacOs and Linux terminals do. 
In Windows, I read that Hyperterm responds to ANSI codes. Realterm responds to most
codes, but will not change the color of the foreground.

The accompanying example file illustrates the library's use.

Copyright (c) 2009 Bruce Robertson, VE9QRP. All rights reserved.

This library is free software; you can redistribute it and/or modify it under
the terms of the GNU Lesser General Public License as published by the Free
Software Foundation; either version 2.1 of the License, or (at your option)
any later version.

This library is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
details.

You should have received a copy of the GNU Lesser General Public License along
with this library; if not, write to the Free Software Foundation, Inc., 51
Franklin St, Fifth Floor, Boston, MA 02110-1301 USA

The latest version of this software is available at
*/

#include "Ansiterm.h"


Ansiterm::Ansiterm()
{
}


void Ansiterm::home()
{
 preamble();
 Serial.write('H');
}


void Ansiterm::xy(int x, int y)
{
 preamble();
 Serial.print(y,DEC);
 Serial.write(';');
 Serial.print(x, DEC);
 Serial.write('H');
}


void Ansiterm::up(int x)
{
 preambleAndNumberAndValue(x,'A');
}


void Ansiterm::down(int x)
{
 preambleAndNumberAndValue(x,'B');
}


void Ansiterm::forward(int x)
{
 preambleAndNumberAndValue(x,'C');
}


void Ansiterm::backward(int x)
{
 preambleAndNumberAndValue(x,'D');
}


void Ansiterm::eraseLine()
{
 preamble();
 Serial.write('2');
 Serial.write('K');
}


void Ansiterm::eraseScreen()
{
 preamble();
 Serial.write('1');
 Serial.write('J');
}


void Ansiterm::setBackgroundColor(int color)
{
 setAttribute(color + 40);
}


void Ansiterm::setForegroundColor(int color)
{
 setAttribute(color + 30);
}


void Ansiterm::boldOn()
{
 setAttribute(BOLD_ON);
}


void Ansiterm::boldOff()
{
 setAttribute(BOLD_OFF);
}


void Ansiterm::italicsOn()
{
 setAttribute(ITALICS_ON);
}


void Ansiterm::italicsOff()
{
 setAttribute(ITALICS_OFF);
}


void Ansiterm::underlineOn()
{
 setAttribute(UNDERLINE_ON);
}


void Ansiterm::underlineOff()
{
 setAttribute(UNDERLINE_OFF);
}


void Ansiterm::strikethroughOn()
{
 setAttribute(STRIKETHROUGH_ON);
}


void Ansiterm::strikethroughOff()
{
 setAttribute(STRIKETHROUGH_OFF);
}


void Ansiterm::inverseOn()
{
 setAttribute(INVERSE_ON);
}


void Ansiterm::inverseOff()
{
 setAttribute(INVERSE_OFF);
}


void Ansiterm::reset()
{
 setAttribute(RESET);
}


void Ansiterm::defaultBackground()
{
 setAttribute(DEFAULT_BACKGROUND);
}


void Ansiterm::defaultForeground()
{
 setAttribute(DEFAULT_FOREGROUND);
}


void Ansiterm::fill(int x1, int y1, int x2, int y2)
{
 for (int x = x1; x <= x2; x++)
 {
   for (int y = y1; y <= y2; y++)
   {
     xy(x,y);
     Serial.print(' ');
   }
 }
}


/* private functions */
void Ansiterm::preamble()
{
 Serial.write(ESCAPE);
 Serial.write(BRACE);
}


void Ansiterm::preambleAndNumberAndValue(int x, char v)
{
 preamble();
 Serial.print(x,DEC);
 Serial.write(v);
}


void Ansiterm::setAttribute(int a)
{
 preambleAndNumberAndValue(a, 'm');
}